packages/shopify-codemod/transforms/iife-to-ternary-expression.js

```diff
@@ -0,0 +1,48 @@
+export default function iifeToTernaryExpression({source}, {jscodeshift: j}, {printOptions = {}}) {
+  function isImmediatelyInvokedFunctionExpression(expressionPath) {
+    let expression = expressionPath.node.expression
+    return expression.type === 'CallExpression' && expression.callee.type === 'ArrowFunctionExpression'
+  }
+
+  function containsOnlyIfStatement(functionNode) {
+    return nodeIsBlockStatementContainingOnly(functionNode.body, 'IfStatement');
+  }
+
+  function onlyReturnsOneOfTwoValues(ifStatementNode) {
+    return nodeIsBlockStatementContainingOnly(ifStatementNode.consequent, 'ReturnStatement') &&
+           nodeIsBlockStatementContainingOnly(ifStatementNode.alternate, 'ReturnStatement');
+  }
+
+  function nodeIsBlockStatementContainingOnly(node, childType) {
+    return node.type === 'BlockStatement' &&
+           node.body.length === 1 &&
+         node.body[0].type === childType;
+  }
+
+  function isConvertibleExpression(path) {
+    return isImmediatelyInvokedFunctionExpression(path) &&
+           containsOnlyIfStatement(path.node.expression.callee) &&
+           onlyReturnsOneOfTwoValues(path.node.expression.callee.body.body[0]);
+  }
+
+  const sourceAST = j(source);
+
+  sourceAST
+    .find(j.ExpressionStatement)
```

Comment:
Anchoring the search on `ExpressionStatement` will sklip a bunch of transformable code.

This transform should be shooting for stuff like this (actual `shopify/shopify` decaffed code!):
```js
const data = {
  url: (() => {
    if (title) {
      return Shopify.routes.app(this.slug).html;
    } else {
      return null;
    }
  })(),

  name: this.name,
};
```

The result should look something like this:
```js
const data = {
  url: (title ? Shopify.routes.app(this.slug).html : null),
  name: this.name,
};
```